docs/tox.ini

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+[tox]
+work_dir = {toxinidir}/../.tox
+requires =
+    tox>=4
+envlist = docs
+
+[testenv]
+basepython = python3
+setenv =
+  PY_COLORS=1
+
+[testenv:docs]
+description = Build the Sphinx docs
+deps = pip-tools
+commands_pre =
+    pip-sync {toxinidir}/.sphinx/requirements.txt
```

Comment:
What do you think about using `uv` here? It's still relatively new, but seems pretty stable, and we're using `ruff` elsewhere.

A cold build with `pip-tools` takes 23.4s on my machine, and a warm build 1.5s (although it fails in the sphinx-build stage). A cold build with `uv pip` takes 2.5s, and a warm one 0.3s (again failing).

Reply:
The `uv` tool shows a lot of promise, but I would prefer we didn't experiment with Python packaging here, and just used the tried-and-true for now. We can always reevaluate later.